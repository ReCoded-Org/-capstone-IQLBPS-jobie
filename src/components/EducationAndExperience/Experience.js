import React, { useState } from "react";
import { useTranslation } from "react-i18next";

const Experience = () => {
  const { t } = useTranslation();
  const [isShowForm, setIsShowForm] = useState(false);

  function handleShowForm() {
    setIsShowForm(true);
  }
  function handleRemoveShowForm() {
    setIsShowForm(false);
  }

  return (
    <div className="bg-gray-200">
      <div className="pt-4 pb-10">
        <h1 className="pb-5">{t("work-experience")}</h1>
        <button
          type="button"
          onClick={handleShowForm}
          className="bg-[#154360] hover:bg-[#226086] text-white font-bold py-2 px-4 rounded-full  left-10 top-10 "
        >
          {t("add-item-experience")}
        </button>
      </div>
      {isShowForm && (
        <div className="pt-4">
          <h1 className="pb-5">{t("experience-item")}</h1>
          <button
            type="button"
            onClick={handleRemoveShowForm}
            className="bg-transparent hover:bg-[#154360] text-[#154360] font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded-full"
          >
            {t("remove-experience")}
          </button>
          <div className="bg-gray-200 flex flex-col justify-center">
            <form className="max-w-[400px] w-full mx-auto rounded-lg bg-gray-200 p-8 px-8">
              <div className="flex flex-col text-black-400 py-2">
                <lable> {t("company")} </lable>
                <input
                  type="text"
                  className="rounded-lg bg-white mt-2 p-2 focus:border-blue-500  focus:outline-none"
                />
              </div>

              <div className="flex flex-col text-black-400 py-2">
                <lable> {t("location-experience")}</lable>
                <input
                  type="text"
                  required
                  className="rounded-lg bg-white mt-2 p-2 focus:border-blue-500  focus:outline-none"
                />
              </div>
              <div className="flex flex-col text-black-400 py-2">
                <lable>{t("type-of-employment-exp")}</lable>
                <input
                  type="text"
                  required
                  className="rounded-lg bg-white mt-2 p-2 focus:border-blue-500  focus:outline-none"
                />
              </div>
              <div className="flex flex-col text-black-400 py-2">
                <lable className=""> {t("position")}</lable>
                <input
                  type="text"
                  required
                  className="rounded-lg bg-white mt-2 p-2 focus:border-blue-500  focus:outline-none"
                />
              </div>
              <div className="flex flex-col text-black-400 py-2">
                <lable>{t("date-exp")}</lable>
                <input
                  type="date"
                  required
                  className="rounded-lg bg-white mt-2 p-2 focus:border-blue-500  focus:outline-none"
                />
              </div>
            </form>
          </div>
        </div>
      )}
    </div>
  );
};

export default Experience;
