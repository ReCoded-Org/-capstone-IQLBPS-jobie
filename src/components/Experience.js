import React ,{ useState } from 'react'

const Experience = () => {
  const [isShowForm,setIsShowForm] = useState(false)
     

   function handleShowForm(){
     setIsShowForm(true)
   }
    function handleRemoveShowForm(){
      setIsShowForm(false)
    }
  
  return (
    <div className='bg-blue-100 '>  
                <div  className='relative h-32 w-32'>
                <h1 className='  pt-5 pl-5'>Work Experience</h1>
                 <button 
                       type="button" 
                       onClick={handleShowForm}
                       className="bg-blue-600 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded-full left-10 top-10  mt-5 ml-5"
          
                      >Add Item</button>
                </div>
           {isShowForm &&   
             <div>
                   <span>Experience Item</span>
                   <button 
                   type="button"
                    onClick={handleRemoveShowForm}
                    className="bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded-full"
                    >Remove</button>
                 <div className='bg-blue-100 grid grid-cols-3 divide-x'>
            <form  className='grid grid-cols-3 divide-x-[3px]' >
          <div>
               <lable> Company </lable>
               <input type="text"/>
          </div>


          <div>
               <lable> Location</lable>
                <input type="text" required/>

          </div>
          <div> 
              <lable>Type of Emploment</lable>
              <input type="text" required />

          </div>
          <div>
                <lable> Postion</lable>
                <input type="text" required />
          </div>
          <div> 
              <lable>Date</lable>
              <input type="date" required />
        </div>
            
          
        </form>
        </div>
             </div>
           
           }
       
    </div>
  )
}

export default Experience

























