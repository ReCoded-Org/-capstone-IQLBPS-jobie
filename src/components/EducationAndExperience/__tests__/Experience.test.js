import renderer from 'react-test-renderer';
import  Experience from "../Experience"
import React from "react"

it('renders Experience commponent correctly', () => {
  const tree = renderer
    .create( <Experience/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});