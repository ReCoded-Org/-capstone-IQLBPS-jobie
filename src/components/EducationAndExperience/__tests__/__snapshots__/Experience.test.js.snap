// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Experience commponent correctly 1`] = `
<div
  className="bg-gray-200"
>
  <div
    className="pt-4 pb-10"
  >
    <h1
      className="pb-5"
    >
      Work Experience
    </h1>
    <button
      className="bg-[#154360] hover:bg-[#226086] text-white font-bold py-2 px-4 rounded-full  left-10 top-10 "
      onClick={[Function]}
      type="button"
    >
      Add Item
    </button>
  </div>
</div>
`;
